### PR TITLE
multus-cni: init at 3.7.1

### PR DESCRIPTION
**Pull request by onixie**
###### Motivation for this change
Multus CNI is a container network interface (CNI) plugin for Kubernetes that enables attaching multiple network interfaces to pods.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/multus-cni/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  preInstall = ''
+      mv $GOPATH/bin/cmd $GOPATH/bin/multus
+  '';
+
+  vendorSha256 = null;
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Multus CNI is a container network interface (CNI) plugin for Kubernetes that enables attaching multiple network interfaces to pods. ";
+    homepage = https://github.com/k8snetworkplumbingwg/multus-cni;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/multus-cni/default.nix:30]:
```suggestion
    homepage = "https://github.com/k8snetworkplumbingwg/multus-cni";
```

File: pkgs/applications/networking/cluster/multus-cni/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  in [
+    "-ldflags=-s -w -X '${multus}.version=v${version}' -X '${multus}.commit=${commit}'"
+  ];
+
+  preInstall = ''
+      mv $GOPATH/bin/cmd $GOPATH/bin/multus
+  '';
+
+  vendorSha256 = null;
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/multus-cni/default.nix:26]: Why did you disable tests?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/multus-cni/default.nix (hunk 2 above)
Comment (onixie) [reply pkgs/applications/networking/cluster/multus-cni/default.nix:26]: The test suite from multus-cni assumes an existing k8s cluster ready at local. This assumption does not always hold  especially when we can only build the package, such as using nixops in a non-nixos linux distro.

BTW, I've been verifying multus-cni (although the master rev) in a real use case for 1+ years in a project.
https://github.com/onixie/nikops/blob/master/src/addons/multus-cni.nix

Comment (SuperSandro2000) [reply pkgs/applications/networking/cluster/multus-cni/default.nix:26]: Please add a comment that the tests require a k8s cluster.

Comment (onixie) [reply pkgs/applications/networking/cluster/multus-cni/default.nix:28]: Sure, comment added.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash the 3 package commits together.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -7451,6 +7451,12 @@
     githubId = 1538622;
     name = "Michael Reilly";
   };
+  onixie = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7454]: Please put this into a separate commit with the message ``maintainers: add onixie`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
